test/nodes/test_retriever.py

```diff
@@ -47,22 +47,18 @@
         ("embedding", "milvus"),
         ("bm25", "elasticsearch"),
         ("bm25", "memory"),
-        ("es_filter_only", "elasticsearch"),
         ("tfidf", "memory"),
     ],
     indirect=True,
 )
 def test_retrieval_without_filters(retriever_with_docs: BaseRetriever, document_store_with_docs: BaseDocumentStore):
-    if not isinstance(retriever_with_docs, (BM25Retriever, FilterRetriever, TfidfRetriever)):
+    if not isinstance(retriever_with_docs, (BM25Retriever, TfidfRetriever)):
         document_store_with_docs.update_embeddings(retriever_with_docs)
 
-    # NOTE: FilterRetriever simply returns all documents matching a filter,
-    # so without filters applied it does nothing
-    if not isinstance(retriever_with_docs, FilterRetriever):
```

Comment:
In https://github.com/deepset-ai/haystack/pull/3561#discussion_r1023849863, we decided to test retrieval with filters and without filters in two separate tests instead of one complex test.

So these checks and comments about filters in `test_retrieval_without_filters` were left over from the previous implementation.